aerogear/unifiedpush-operator
# Do not try to watch Enmasse resources if enmasse doesn't support watches

## Motivation
Enmasse 0.30.2 has an incorrect implementation of the watch operation which causes the operator to spam the enmasse service for a list of their resources.  This can cause issues with the enmasse service.  Enmasse 0.31 fixes watches, but hasn't been released yet.

https://issues.redhat.com/browse/AEROGEAR-10176

## What
This PR fixes that issue by detecting an incorrect implementation of enmasse watches.  If an incorrect implementation is detected, then AMQ features will not be enabled in UPS.  Additionally watches for these resources will not be created.

## Why
The failing watch causes the operator to requeue immediately and try again.

## How
There is a method to check if enmasse supports watches correctly.  *The result is cached*.  However, in the future we should be able to remove this once enmasse 0.31+ is more widely adopted

## Verification Steps
 
1. Have a cluster with enmasse 0.30.x running
2. Install the UPS operator
3. Check the operator logs, there should not be errors regarding enmasse
4. Create UnifiedPush with a broker
5. Address, AddressSpace, MessagingUser, etc should *NOT* be created, but UPS should still deploy
6. Upgrade to enmasse 0.31.x (it is in rc right now)
7. Confirm UPS with AMQ deploys successfully

## Checklist:

- [ x ] Code has been tested locally by PR requester
